cmd/kubelet/app/server.go

```diff
@@ -1100,7 +1100,9 @@ func RunKubelet(kubeServer *options.KubeletServer, kubeDeps *kubelet.Dependencie
 	}
 	podCfg := kubeDeps.PodConfig
 
-	rlimit.RlimitNumFiles(uint64(kubeServer.MaxOpenFiles))
+	if err := rlimit.RlimitNumFiles(uint64(kubeServer.MaxOpenFiles)); err != nil {
```

Comment:
Can you share more about the situations in which setting the `Rlimit` could fail?

Also, is there any reason that we need to run this check after `createAndInitKubelet`? If not, can I propose we move it to the top of this function, so that if setting the Rlimit does fail, we don't need to wait to create and init the Kubelet to find that out?

Reply:
It is a low-level system call associated with the operating system.
I'm not sure what circumstances might cause this call to fail, but I think it's worth doing something to make the code more robust.

I agree with you that we should move it to the top of this function if possible. 
Thanks @mattjmcnaughton

Reply:
Yeah, plus one I'm also not really sure what could cause this system call to fail... however, if it does feel, I agree we want to know about it.